From 33b92fc51641a5c4c58935002b915b4189ab113b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:43 -0400
Subject: [usb] usbtest: support bos descriptor test for usb 3.0

Message-id: <1410887463-20674-152-git-send-email-dzickus@redhat.com>
Patchwork-id: 95244
O-Subject: [RHEL7 PATCH 151/591] usb: usbtest: support bos descriptor test for usb 3.0
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9d3bd7684645834ede59d285af1d70ccabee9bf3
Author: Huang Rui <ray.huang@amd.com>
Date:   Mon Oct 28 23:31:32 2013 +0800

    usb: usbtest: support bos descriptor test for usb 3.0

    In Test 9 of usbtest module, it is used for performing chapter 9 tests N
    times. This patch adds to support getting bos descriptor test scenario for
    USB 3.0.

    Signed-off-by: Huang Rui <ray.huang@amd.com>
    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index 1fdd88f..e7235d5 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -689,6 +689,19 @@ static int ch9_postconfig(struct usbtest_dev *dev)
   return (retval < 0) ? retval : -EDOM;
  }
 
+ /*
+  * there's always [9.4.3] a bos device descriptor [9.6.2] in USB
+  * 3.0 spec
+  */
+ if (le16_to_cpu(udev->descriptor.bcdUSB) >= 0x0300) {
+  retval = usb_get_descriptor(udev, USB_DT_BOS, 0, dev->buf,
+    sizeof(*udev->bos->desc));
+  if (retval != sizeof(*udev->bos->desc)) {
+   dev_err(&iface->dev, "bos descriptor --> %d\n", retval);
+   return (retval < 0) ? retval : -EDOM;
+  }
+ }
+
  /* there's always [9.4.3] at least one config descriptor [9.6.3] */
  for (i = 0; i < udev->descriptor.bNumConfigurations; i++) {
   retval = usb_get_descriptor(udev, USB_DT_CONFIG, i,
-- 
1.7.1